src/rai_bench/rai_bench/tool_calling_agent/predefined/basic_tasks.py

```diff
@@ -20,164 +20,375 @@
+        "parameters.0.value.type": 2,
+        "parameters.0.value.integer_value": DEFAULT_FPS,
+    },
+)
+set_o3de_fps_atomically_subtask_1 = CheckServiceFieldsToolCallSubTask(
+    expected_tool_name="call_ros2_service",
+    expected_service=O3DE_SET_PARAMS_ATOMICALLY,
+    expected_service_type=SET_PARAMETERS_ATOMICALLY_TYPE,
+    expected_fields={
+        "parameters.0.name": "fps",
+        "parameters.0.value.type": 2,
+        "parameters.0.value.integer_value": DEFAULT_FPS,
+    },
+)
+
+set_grounded_sam_subtask_2 = CheckServiceFieldsToolCallSubTask(
+    expected_tool_name="call_ros2_service",
+    expected_service=GROUNDED_SAM_SET_PARAMS,
+    expected_service_type=SET_PARAMETERS_TYPE,
+    expected_fields={
+        "parameters.0.name": "confidence_threshold",
+        "parameters.0.value.type": 3,
+        "parameters.0.value.double_value": SAM_CONFIDENCE_2,
+    },
+)
+set_grounded_sam_atomically_subtask_2 = CheckServiceFieldsToolCallSubTask(
+    expected_tool_name="call_ros2_service",
+    expected_service=GROUNDED_SAM_SET_PARAMS_ATOMICALLY,
+    expected_service_type=SET_PARAMETERS_ATOMICALLY_TYPE,
+    expected_fields={
+        "parameters.0.name": "confidence_threshold",
+        "parameters.0.value.type": 3,
+        "parameters.0.value.double_value": SAM_CONFIDENCE_2,
+    },
+)
+
+set_grounding_dino_subtask_2 = CheckServiceFieldsToolCallSubTask(
+    expected_tool_name="call_ros2_service",
+    expected_service=GROUNDING_DINO_SET_PARAMS,
+    expected_service_type=SET_PARAMETERS_TYPE,
+    expected_fields={
+        "parameters.0.name": "confidence_threshold",
+        "parameters.0.value.type": 3,
+        "parameters.0.value.double_value": DINO_CONFIDENCE_2,
+    },
+)
+set_grounding_dino_atomically_subtask_2 = CheckServiceFieldsToolCallSubTask(
+    expected_tool_name="call_ros2_service",
+    expected_service=GROUNDING_DINO_SET_PARAMS_ATOMICALLY,
+    expected_service_type=SET_PARAMETERS_ATOMICALLY_TYPE,
+    expected_fields={
+        "parameters.0.name": "confidence_threshold",
+        "parameters.0.value.type": 3,
+        "parameters.0.value.double_value": DINO_CONFIDENCE_2,
+    },
+)
+
+set_o3de_fps_subtask_2 = CheckServiceFieldsToolCallSubTask(
+    expected_tool_name="call_ros2_service",
+    expected_service=O3DE_SET_PARAMS,
+    expected_service_type=SET_PARAMETERS_TYPE,
+    expected_fields={
+        "parameters.0.name": "fps",
+        "parameters.0.value.type": 2,
+        "parameters.0.value.integer_value": FPS_2,
+    },
+)
+set_o3de_fps_atomically_subtask_2 = CheckServiceFieldsToolCallSubTask(
+    expected_tool_name="call_ros2_service",
+    expected_service=O3DE_SET_PARAMS_ATOMICALLY,
+    expected_service_type=SET_PARAMETERS_ATOMICALLY_TYPE,
+    expected_fields={
+        "parameters.0.name": "fps",
+        "parameters.0.value.type": 2,
+        "parameters.0.value.integer_value": FPS_2,
+    },
+)
+
+
+delete_entity_subtask1 = CheckServiceFieldsToolCallSubTask(
```

Comment:
do we need all these tasks by name?
Wouldn't it be possible to generate these tasks with some factory function with much less code?